### PR TITLE
issue 21 -- add crd validation for cpu and memory

### LINKED ISSUE #21: Add more CRD validation--specifically for the cpu and memory values
Under spec.resources.limits and spec.resources.requests are values for CPU and memory (refer to https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu  & #meaning-of-memory for details on acceptable values).

### PR DESCRIPTION
**Pull request by ginnick**
fixes #21

### CODE REVIEW

File: olm/open-liberty-crd.yaml [hunk 1]
```diff
@@ -363,8 +363,10 @@ spec:
                   properties:
                     cpu:
                       type: string
+                      pattern: '^[0-9]+m?$'
```
Comment (mezarin) [inline olm/open-liberty-crd.yaml]: This line does appear to handle fractions (i.e. 0.1 = 100m).
Doc: ""The upper limit of CPU core. Specify integers, fractions (e.g. 0.5), or millicores values(e.g. 100m, where 100m is equivalent to .1 core).""

File: olm/open-liberty-crd.yaml [hunk 2]
```diff
@@ -373,8 +375,10 @@ spec:
...
                     memory:
                       type: string
+                      pattern: '^[0-9]+([kKmMgGtTpPeE][i]?|[e][0-9]*)?$'
```
Comment (mezarin) [inline olm/open-liberty-crd.yaml]: This looks fine, but just one observation that the doc (https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/) states that these suffixes be used: E, P, T, G, M, K. Your expression allows for lower case values such as 100gi, or 10e (10 exa bytes).

File: olm/open-liberty-crd.yaml (hunk 1 above)
Comment (ginnick) [reply olm/open-liberty-crd.yaml]: OK...will update to support fractions.

File: olm/open-liberty-crd.yaml (hunk 2 above)
Comment (ginnick) [reply olm/open-liberty-crd.yaml]: Will update to remove the mixed-case support.

**Pull request merged**